apps/microsoft-teams/frontend/src/components/config/AccessSection/AccessSection.tsx

```diff
@@ -1,57 +1,128 @@
+  const sdk = useSDK();
+  const { logout, login, teamsAppInfo, teamsAppLink, description } = accessSection;
 
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
-    dispatch({
-      type: actions.UPDATE_TENANT_ID,
-      payload: e.target.value,
+  const handleLogin = async () => {
+    try {
+      const authResult = await instance.loginPopup(loginRequest);
+      if (!isAppInstalled) {
+        await customApi.saveConfiguration({ ...parameters, tenantId: authResult.tenantId });
+        dispatch({
+          type: actions.UPDATE_TENANT_ID,
+          payload: authResult.tenantId,
+        });
+      }
+    } catch (e) {
+      const message = e instanceof Error ? e.message : 'Failed to authenticate with Microsoft';
+      sdk.notifier.error(message);
+      console.log(e);
+    }
+  };
+
+  const handleLogout = () => {
+    ModalLauncher.open(({ isShown, onClose }) => {
+      return (
+        <DisconnectModal
+          isShown={isShown}
+          handleCancel={() => {
+            onClose(true);
+          }}
+          handleDisconnect={async () => {
+            onClose(true);
+            await instance.logoutPopup({
+              mainWindowRedirectUri: '/',
+            });
+          }}
+        />
+      );
     });
   };
 
-  return (
-    <Box className={styles.box}>
-      <Heading>{headerSection.title}</Heading>
-      <Paragraph>{headerSection.description}</Paragraph>
-      <hr className={styles.splitter} />
-      <Subheading>{accessSection.title}</Subheading>
-      <FormControl data-test-id="tenant-id-section">
-        <FormControl.Label>{accessSection.fieldName}</FormControl.Label>
-        <TextInput value={tenantId} type="text" name="tenantId" onChange={handleChange} />
-      </FormControl>
-      <Flex marginBottom="spacingS" alignItems="center" className={styles.logo}>
-        <TeamsLogo />
-        <Box marginLeft="spacingXs">
-          <HyperLink
-            body={accessSection.teamsAppInfo}
-            substring={accessSection.teamsAppLink}
-            // TODO: update link to app documentation
-            href={'https://www.contentful.com/help/apps-at-contentful/'}
-          />
-        </Box>
-      </Flex>
-    </Box>
-  );
+  const accessComponent = () => {
+    // Display if not authorized and if logout is not in progress
+    if (!accounts.length && inProgress !== 'logout') {
+      return (
+        <>
+          <Paragraph>{description}</Paragraph>
+          <Box>
+            <Button
+              onClick={() => handleLogin()}
+              isDisabled={inProgress === 'login'}
+              isLoading={inProgress === 'login'}>
+              {login}
+            </Button>
+          </Box>
+        </>
+      );
+    } else {
+      return (
+        <>
+          <Card padding="large">
+            <Flex justifyContent="space-between">
+              <Flex flexDirection="column">
+                <Subheading marginBottom="none">{accounts[0]?.name}</Subheading>
```

Comment:
just my curiosity, is the accounts type just always an array, which is why we have to index it here?

Reply:
Yes, accounts comes from the `useMsal` hook and the type is `AccountInfo[]`. So it's an array of authenticated accounts with info like name, username, tenantid, etc. https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/src/MsalContext.ts#L18